### PR TITLE
Allow the pull command to fetch source string content

### PR DESCRIPTION
**Pull request by stelabouras**
Given that CDS now allows the pulling of source content, changes had to
be made to enable that functionality on the SDK.

This commit changes the `CDSHandler` initialization by allowing it to
accept all application locales that have been declared by the developer
during the SDK initialization, which include the source locale.

The translation logic has been also updated, so that there is no special
treatment for the source locale; the string key is generated regardless
of the locale and the same strategy is always used: The key is checked
against the cache (which will now also contain the source locale) and
if the key is not found in the cache and the requested locale is not the
source one, then the source locale is used to hit the cache again. If
finally the string cannot be found in the cache even for the source
locale, then the bundled resources are queried (via the
`BypassLocalizer` class) and the missing policy is used.

A new test has been added to test whether the source locale is fetched
from the cache for a simple and a plural approach.

### CODE REVIEW

**Review by dbendilas — CHANGES_REQUESTED** [review]
If I'm reading the code correctly, the implementation is not the desired one. This is what we need:

- If translating in source locale, check the cache for a fresh source translation
- If it exists, use that, otherwise use the source string as is (not sure I remember the case for `bypassLocalizer`)

- If translating in target locale, check the cache for a fresh target translation
- If it exists, use that, otherwise delegate the logic to the missing policy.

**Comment by stelabouras** [comment]
> If I'm reading the code correctly, the implementation is not the desired one. This is what we need:
> 
> **Source locale**
> 
>     * Check the cache for a source translation
> 
>     * If it exists, use that, otherwise use the source string as is (not sure I remember the case for `bypassLocalizer`)
> 
> 
> **Target locale**
> 
>     * Check the cache for a target translation
> 
>     * If it exists, use that, otherwise delegate the logic to the missing policy.

Gotcha. So for target locales, if the target translation is NOT found in cache, then the logic shouldn't attempt to load the source translation from cache, is that correct?

Also if -in both cases (either source or target locale)- the translation is NOT found in cache, then the bypass localizer is used before applying the missing policy, in order to avoid cases where the source string is not a human readable key (e.g. "my_localized_key") and the bundle contains the translation in a `.strings` file, so that the UI won't reveal any keys to the user, but the bundled translation. If the default missing policy is selected (`TXSourceStringPolicy`) then this won't result into displaying weird characters on the UI by default. Is that what we want here for both cases? Or should we treat the source locale case differently?

File: Sources/Transifex/Core.swift [hunk 1]
```diff
@@ -226,36 +224,45 @@ class NativeCore : TranslationProvider {
...
-            let key = txGenerateKey(sourceString: sourceString,
-                                    context: context)
-            translationTemplate = cache.get(key: key,
-                                            localeCode: localeToRender)
-            if !String.containsTranslation(translationTemplate) {
-                let bypassedString = self.bypassLocalizer.get(sourceString: sourceString,
-                                                              params: params)
-                
-                return missingPolicy.get(sourceString: bypassedString)
+        /// If the source string is not found in the cache, use the bypass localizer to look it up on the
```
Comment (dbendilas) [inline Sources/Transifex/Core.swift:236]: `If the source string` -> should be `If the string`

Comment (stelabouras) [reply Sources/Transifex/Core.swift]: Done!

**Review by dbendilas — APPROVED** [review]
Looks great!

**Pull request merged**